File: springboot-backend/src/main/java/backend/dto/request/OrderRequest.java
package backend.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.ArrayList; // Thêm import này
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderRequest {

    @JsonProperty("shippingAddress")
    private String shippingAddress = "";

    @JsonProperty("phoneNumber")
    private String phoneNumber = "";

    @JsonProperty("paymentMethod")
    private String paymentMethod = "COD";

    @JsonProperty("items")
    private List<CartItemRequest> items = new ArrayList<>();
}

File: Fontend/app/src/main/java/com/example/data/dto/request/OrderRequest.kt
package com.example.data.dto.request

import com.google.gson.annotations.SerializedName

data class OrderRequest(
    @SerializedName("shippingAddress")
    val shippingAddress: String,

    @SerializedName("phoneNumber")
    val phoneNumber: String,

    @SerializedName("paymentMethod")
    val paymentMethod: String,

    @SerializedName("items")
    val items: List<CartItemRequest>
)